src/stk500v2.c

```diff
@@ -3718,20 +3724,22 @@ static void stk500v2_print_parms1(const PROGRAMMER *pgm, const char *p, FILE *fp
     break;
 
   case PGMTYPE_STK600:
-    stk500v2_getparm2(pgm, PARAM2_AREF0, &varef);
-    fmsg_out(fp, "%sVaref 0         : %.2f V\n", p, varef / 100.0);
-    stk500v2_getparm2(pgm, PARAM2_AREF1, &varef);
-    fmsg_out(fp, "%sVaref 1         : %.2f V\n", p, varef / 100.0);
+    if (pgm->extra_features & HAS_VAREF_ADJ) {
+      stk500v2_getparm2(pgm, PARAM2_AREF0, &varef);
+      fmsg_out(fp, "%sVaref 0         : %.2f V\n", p, varef / 100.0);
+      stk500v2_getparm2(pgm, PARAM2_AREF1, &varef);
+      fmsg_out(fp, "%sVaref 1         : %.2f V\n", p, varef / 100.0);
+    }
     stk500v2_getparm2(pgm, PARAM2_SCK_DURATION, &sck_stk600);
-    fmsg_out(fp, "%sSCK period      : %.2f us\n", p,
-	    (float) (sck_stk600 + 1) / 8.0);
-    stk500v2_getparm2(pgm, PARAM2_CLOCK_CONF, &clock_conf);
-    oct = (clock_conf & 0xf000) >> 12u;
-    dac = (clock_conf & 0x0ffc) >> 2u;
-    f = pow(2, (double)oct) * 2078.0 / (2 - (double)dac / 1024.0);
-    f = f_to_kHz_MHz(f, &unit);
-    fmsg_out(fp, "%sOscillator      : %.3f %s\n",
-            p, f, unit);
+    fmsg_out(fp, "%sSCK period      : %.2f us\n", p, (float) (sck_stk600 + 1) / 8.0);
```

Comment:
Small comment. The pre-existing cast `(float)` is superfluous: Owing to the division by 8.0 the nominator is cast to `double` anyway, and `%f` always expects a double. Actually, all `(float)` casts in stk500v2.c and jtag3.c are superfluous by the same reasoning.